src/Mesh/Interpolation.jl

```diff
@@ -40,45 +52,69 @@ struct InterpolationBrick{
+    x1g::FTV
+    "Interpolation grid in x2 direction"
     x2g::FTV
+    "Interpolation grid in x3 direction"
     x3g::FTV
-
-    ξ1::FTVD # unique ξ1 coordinates of interpolation points within each element
-    ξ2::FTVD # unique ξ2 coordinates of interpolation points within each element
-    ξ3::FTVD # unique ξ3 coordinates of interpolation points within each element
-
-    flg::UI8AD # flags when ξ1/ξ2/ξ3 interpolation point matches with a GLL point
-
-    fac::FTVD # normalization factor
-
-    x1i::UI16VD # interpolation grid indices of interpolation points within each element
+    "unique ξ1 coordinates of interpolation points within each spectral element"
+    ξ1::FTVD
+    "unique ξ2 coordinates of interpolation points within each spectral element"
+    ξ2::FTVD
+    "unique ξ3 coordinates of interpolation points within each spectral element"
+    ξ3::FTVD
+    "flags when ξ1/ξ2/ξ3 interpolation point matches with a GLL point"
+    flg::UI8AD
+    "normalization factor"
+    fac::FTVD
+    "x1 interpolation grid index of interpolation points within each element on the local process"
+    x1i::UI16VD
+    "x2 interpolation grid index of interpolation points within each element on the local process"
     x2i::UI16VD
+    "x3 interpolation grid index of interpolation points within each element on the local process"
     x3i::UI16VD
-
-    offset::TVD  # offsets for each element for v
-
-    m1_r::FTVD  # GLL points
-    m1_w::FTVD  # GLL weights
-    wb::FTVD    # Barycentric weights
+    "Offsets for each element"
+    offset::TVD    # offsets for each element for v
+    "GLL points"
+    m1_r::FTVD
+    "GLL weights"
+    m1_w::FTVD
+    "Barycentric weights"
+    wb::FTVD
     # MPI setup for gathering interpolated variable on proc # 0
+    "Number of interpolation points on each of the processes"
     Nel_all::I32V
 
-    x1i_all::UI16V
-    x2i_all::UI16V
-    x3i_all::UI16V
+    "x1 interpolation grid index of interpolation points within each element on all processes stored only on proc 0"
+    x1i_all::UI16VD
+    "x2 interpolation grid index of interpolation points within each element on all processes stored only on proc 0"
+    x2i_all::UI16VD
+    "x3 interpolation grid index of interpolation points within each element on all processes stored only on proc 0"
+    x3i_all::UI16VD
     #--------------------------------------------------------
+    """
+        InterpolationBrick(grid::DiscontinuousSpectralElementGrid{FT}, xbnd::Array{FT,2}, xres) where FT <: AbstractFloat
+    This interpolation structure and the corresponding functions works for a brick, where stretching/compression happens only along the x, y & z axis.
+    Here x1 = X1(ξ1); x2 = X2(ξ2); x3 = X3(ξ3)
+    # input for the inner constructor
+     - `grid` DiscontinousSpectralElementGrid
+     - `xbnd` Domain boundaries in x1, x2 and x3 directions
+     - `xres` Resolution of the interpolation grid in x1, x2 and x3 directions
+    """
```

Comment:
The issue here is that docstrings on inner constructors are ignored.

The easiest solution is to add this to the docstring for the struct itself.

Reply:
Done.